services/CurrentTime/source/CurrentTimeService.cpp

```diff
@@ -45,18 +50,60 @@ void CurrentTimeService::set_event_handler(EventHandler *handler) {
     _current_time_handler = handler;
 }
 
-time_t CurrentTimeService::get_time()
+time_t CurrentTimeService::get_time() const
 {
     time_t epoch_time = time(nullptr);
 
     return epoch_time + _time_offset;
 }
 
-void CurrentTimeService::set_time(time_t host_time)
+void CurrentTimeService::set_time(time_t host_time, uint8_t adjust_reason)
 {
     time_t epoch_time = time(nullptr);
 
+    time_t old_time_offset = _time_offset;
     _time_offset = host_time - epoch_time;
+
+    update_current_time_value(adjust_reason, old_time_offset - _time_offset);
+}
+
+void CurrentTimeService::update_current_time_value(const uint8_t adjust_reason, time_t time_offset_difference) {
+    time_t local_time = get_time();
+
+    struct tm *local_time_tm = localtime(&local_time);
+
+    CurrentTime current_time(local_time_tm);
+
+    current_time.adjust_reason = adjust_reason;
+
+    bool send_update = true;
+    if ((current_time.adjust_reason & (MANUAL_TIME_UPDATE | CHANGE_OF_TIME_ZONE | CHANGE_OF_DST)) == 0) {
```

Comment:
I don't think that is correct, `(EXTERNAL_REFERENCE_TIME_UPDATE | CHANGE_OF_TIME_ZONE)` should trigger an update.

Reply:
CHANGE_OF_TIME_ZONE already triggers an update (default value of send_update is true) and EXTERNAL_REFERENCE_TIME_UPDATE will also send an update but only if 15 minutes have elapsed (as mandated by spec)

Reply:
Damn, I'm an idiot. Thanks @paul-szczepanek-arm .